keylime/common/retry.py

```diff
@@ -0,0 +1,16 @@
+"""
+SPDX-License-Identifier: Apache-2.0
+Copyright 2021 Angelo Ruocco - IBM Research Lab Zurich
+"""
+
+def retry_time(exponential, base, ntries, logger):
+    if exponential:
+        if base > 1:
+            return base**ntries
+        else:
+            try:
+                logger.warning("Base %f incompatible with exponential backoff", base)
+            except:
+                None
+
+    return abs(base)
```

Comment:
Instead of catching all exceptions you can also just check if the logger is there:
```suggestion
def retry_time(exponential, base, ntries, logger=None):
    if exponential:
        if base > 1:
            return base**ntries
        elif logger:
            logger.warning("Base %f incompatible with exponential backoff", base)

    return abs(base)
```

Reply:
Ah yeah of course, fixed!